react/react1/week3/to-do-app/src/Main.jsx

```diff
@@ -0,0 +1,123 @@
+            })
+        )
+    }
+    const cancelUpdate = ()=>{
+        if(!update){
+            alert('No Task to cancel!')
+        }else{
+            setUpdate('');
+        };
+    }
+    const changeTask = (e) => {
+        let newTask = {
+            id: update.id,
+            description: e.target.value,
+            deadline: e.target.value
+        }
+        setUpdate(newTask)
+        //
+    }
+    const updateTask = () => {
+        if(!update){
+            alert('No Task to update!')
+        }else{
+            let filteredTask = listItem.filter(task => task.id !== update.id)
+            let updatedTask = [...filteredTask, update]
+            setListItem(updatedTask)
+            setUpdate('');
+        }
+    }
+    function ToDoListItems(props) {
+        
+
+        return <ul>
+            {props.lists && props.lists.length ? '' : <h3 className="message">Add your ToDo List</h3>}
+            {props.lists.map(list => {
+                return (
+                    <li key={list.id} className={` list-item ${list.completed ? "completed" : " "}`} title="Checked/Not-Checked" >
+                        <button onClick={() => CheckedList(list.id)} className="checked-button"><div className="checked-mark"></div></button>
+                        {list.description} {list.deadline}
+                        <button onClick={() => DeleteList(list.id)} className="delete-button" title="Delete">&times;</button>
+                        <button className="edit-button" onClick={()=> setUpdate({
+                            id:list.id,
+                            description: list.description,
+                            deadline: list.deadline
+                        })}>Edit</button>
+                    </li>
+                )
+            })}
+        </ul>
+    }
+
+    const AddList = (todo, date) => {
+     const ArrayLength = listItem.length
+     if(!todo){
+        alert("Enter your task to do!")
+     }else if(date < today){
+        alert("Pick future date as a Deadline!")
+     }else {
+        const NewListName = {
+            id: ArrayLength + 1,
+            description: todo,
+            deadline: date
+        };
+        const NewList = listItem.concat(NewListName);
+        setListItem(NewList);
+        setInput('')
+        setDeadline(today)
+     }
+       
+    }
+
+    return (<main className="to-do-content">
+        <form>
+            <label >To Do Description:
+                <input type='text' value={input} onChange={(e) => setInput(e.target.value)}
+                />
+            </label>
+            <br></br>
+            <label >To Do Deadline:
+                <input type='date' value={deadline} onChange={(e) => setDeadline(e.target.value)} />
+            </label>
+        </form>
+        <button onClick={() => AddList(input, deadline)} className="add-button">Add ToDo</button>
+        <br></br>
+        <input className="update-input"
+        value={update && update.description}
```

Comment:
you are duplicating the value when you update the question, test it

Reply:
It is because I have added list.deadline as input value 🙈 I will make that changes